### PR TITLE
feat(init): first commit

### PR DESCRIPTION
**Pull request by jbarascut**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  })
+  default = {
+    day_of_week  = 0
+    start_hour   = 0
+    start_minute = 0
+  }
+}
+
+variable "private_dns_zone_name" {
+  description = ""
```
Comment (fsismondi) [inline variables.tf]: should be described

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+resource "azurerm_postgresql_flexible_server_configuration" "this" {
+  for_each = var.postgresql_config
+
+  server_id = azurerm_postgresql_flexible_server.this.id
+  name      = each.key
+  value     = each.value
+}
+
+resource "azurerm_management_lock" "this" {
```
Comment (fsismondi) [inline main.tf]: good you added this

Comment (fsismondi) [reply main.tf]: will it work tho? when you actually need to delete a DB cluster , how do u do this?

**Review by fsismondi — CHANGES_REQUESTED** [review]

File: variables.tf [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+variable "subnet_address_prefixes" {
+  description = "Address prefixes of Subnet to be created."
+  type        = list(string)
+}
+
+variable "backup_retention_days" {
+  description = "Backup retention days for the PostgreSQL Flexible Server."
+  type        = number
+  default     = 35
```
Comment (fsismondi) [inline variables.tf:87]: is this default in camptocamp? how are most of our psql servers configured?

File: variables.tf [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+}
+
+variable "maintenance_window" {
+  description = "Cron maintenance window (default sunday midnight)."
```
Comment (fsismondi) [inline variables.tf:91]: 0,0,0 is sunday midnight in CET ? or local time?

File: variables.tf [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+  })
+  default = {
+    day_of_week  = 0
+    start_hour   = 0
+    start_minute = 0
+  }
+}
+
+variable "private_dns_zone_name" {
+  description = ""
```
Comment (fsismondi) [inline variables.tf]: "The ID of the private DNS zone to create the PostgreSQL Flexible Server. Changing this forces a new PostgreSQL Flexible Server to be created."

**Review by fsismondi — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 3 above)
Comment (jbarascut) [reply variables.tf:87]: Possible range between 7 to 35. 35 is camptocamp default

File: variables.tf (hunk 4 above)
Comment (jbarascut) [reply variables.tf:91]: Server time maybe

**Review by fsismondi — APPROVED** [review]

File: variables.tf [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+  type        = string
+}
+
+variable "virtual_network_pipeline_id" {
+  description = "Virtual Network ID where Subnet will be created."
+  type        = string
+}
+
+variable "name" {
+  description = "Name for the PostgreSQL Flexible Server."
```
Comment (fsismondi) [inline variables.tf]:
```suggestion
  description = "Name for the PostgreSQL Flexible Server. This is also used for the DNS private zone name."
```

**Pull request merged**